crates/mpz-ot-core/src/ferret/config.rs

```diff
@@ -0,0 +1,188 @@
+    n: 9600,
+    k: 1220,
+    t: 600,
+};
+
+/// Ferret configuration.
+#[derive(Clone, Builder)]
+pub struct FerretConfig {
+    /// LPN type.
+    #[builder(default = "LpnType::Uniform")]
+    lpn_type: LpnType,
+    /// Whether to reserve bootstrap COTs.
+    #[builder(default = "true")]
+    reserve_bootstrap: bool,
+    #[builder(setter(custom), default = "Arc::new(default_parameter_selector)")]
+    param_selector: Arc<dyn Fn(LpnType, usize, usize) -> LpnParameters + Send + Sync + 'static>,
+}
+
+impl Debug for FerretConfig {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("FerretConfig")
+            .field("lpn_type", &self.lpn_type)
+            .field("reserve_bootstrap", &self.reserve_bootstrap)
+            .finish_non_exhaustive()
+    }
+}
+
+impl Default for FerretConfig {
+    fn default() -> Self {
+        Self {
+            lpn_type: LpnType::Uniform,
+            reserve_bootstrap: true,
+            param_selector: Arc::new(default_parameter_selector),
+        }
+    }
+}
+
+impl FerretConfigBuilder {
+    /// Configures the LPN parameter selector.
+    ///
+    /// The provided function must have the following signature:
+    ///
+    /// `(LpnType, available, additional) -> LpnParameters`
+    ///
+    /// where `available` is the current number of available COTs and
+    /// `additional` is the number of COTs that still need to be generated.
+    pub fn param_selector<F>(&mut self, f: F) -> &mut Self
+    where
+        F: Fn(LpnType, usize, usize) -> LpnParameters + Send + Sync + 'static,
+    {
+        self.param_selector = Some(Arc::new(f));
+        self
+    }
+}
+
+impl FerretConfig {
+    /// Returns a new `FerretConfigBuilder`.
+    pub fn builder() -> FerretConfigBuilder {
+        FerretConfigBuilder::default()
+    }
+
+    /// Returns `true` if bootstrap COTs should be reserved.
+    pub fn reserve_bootstrap(&self) -> bool {
+        self.reserve_bootstrap
+    }
+
+    /// Returns the LPN type.
+    pub fn lpn_type(&self) -> LpnType {
+        self.lpn_type
+    }
+
+    /// Returns the cost of a bootstrap iteration.
+    pub(crate) fn bootstrap_cost(&self) -> usize {
+        match self.lpn_type {
+            LpnType::Uniform => iteration_cost(self.lpn_type, UNIFORM_PARAMS[0]),
+            LpnType::Regular => iteration_cost(self.lpn_type, REGULAR_PARAMS[0]),
+        }
+    }
+
+    pub(crate) fn select_params(&self, available: usize, additional: usize) -> LpnParameters {
+        (self.param_selector)(self.lpn_type, available, additional)
+    }
+}
+
+fn default_parameter_selector(ty: LpnType, available: usize, additional: usize) -> LpnParameters {
+    let params = match ty {
+        LpnType::Uniform => UNIFORM_PARAMS,
+        LpnType::Regular => REGULAR_PARAMS,
+    };
+
+    // *Assumes the parameters are in ascending order.*
+    for param in params {
+        let cost = iteration_cost(ty, *param);
+        let net = param.t - cost;
```

Comment:
Why is `net = t - cost`?

Reply:
My understanding is that `n` is the number of COTs that will be produced. I think `t` is the number of LPN errors (hamming weight of the error vector). So I think this should be `net = param.n - cost`, no?

Reply:
seems like a bug, i opened an issue https://github.com/privacy-scaling-explorations/mpz/issues/248